src/lib/task.js

```diff
@@ -696,11 +696,20 @@ export class Task extends EventEmitter {
       // we don't wait for the promise to resolve just trigger kill here which
       // will cause run to stop processing the task and give us an error
       // exit code.
-      this.dockerProcess.kill();
-      this.stream.write(fmtErrorLog(
-        'Task timeout after %d seconds. Force killing container.',
-        this.task.payload.maxRunTime
-      ));
+      // But first we kill with SIGINT, giving the process a chance to cleanup
+      setTimeout(() => {
+        this.dockerProcess.kill();
+        this.stream.write(fmtErrorLog(
+          'Task timeout after %d seconds. Force killing container.',
+          this.task.payload.maxRunTime
+        ));
+      }, 5 * 1000);
+      try {
+        this.dockerProcess.container.kill({signal: 'SIGINT'}).catch(err =>
```

Comment:
yes, and it's really only for tasks that cares about this stuff...

It might be something gps will play with in `run-task` to ensure that mercurial cases aren't locked or anything bad...